vm/rust/src/lib.rs

```diff
@@ -244,7 +235,12 @@ pub extern "C" fn cairoVMExecute(
                 );
                 return;
             }
-            Ok(t) => unsafe {
+            Ok(mut t) => unsafe {
+                // we are estimating fee, override actual fee calculation
+                if !charge_fee {
```

Comment:
Can we move this outside the `unsafe` block?

Reply:
done